scripts/datacube.py

```diff
@@ -11,6 +12,14 @@ def check_keys(ds):
     if required_keys > actual_keys:
         raise KeyError("CanRCM4 ensembles is missing keys {}".format(required_keys - actual_keys))
 
+def check_path(data_path):
+    if isinstance(data_path, str) is False:
+        raise ValueError('Path requires str got {}'.format(type(data_path)))
+    if len(glob.glob(data_path)) == 0:
+        raise IndexError('Path provided has no files with \'.nc\' extension')
+    if len(glob.glob(data_path)) <=20:
+        warnings.warn("Path has a low ensemble size with < 20 members")
```

Comment:
Just checking (hah): You mean here to allow the program to carry on despite the small ensemble size? But abort for the other cases?

Reply:
One additional remark: The very top-most level of your script should catch any exceptions that bubble up from lower levels and emit a user friendly message. The messages embedded in the exceptions will be a big part of that, but in general you don't want to subject the user to the gory result of an uncaught exception.

Even better: Create an exception class of your own for these kinds of validation check errors, and catch only those at the top level. Then any unanticipated exceptions still produce the gory but useful-for-debugging stack trace etc. if they occur.